src/CUDA/gpu_misc.cpp

```diff
@@ -34,6 +34,10 @@ cublasHandle_t cublasHandle;
 
 size_t MaxGPUSpineSizeMB;
 int rank;
+int relative_rank;
+int device_group_size;
+int* device_group_numbers;
```

Comment:
It is better to avoid C pointers. Can we used std::vector?

Reply:
Yes, I'll upload it later today.